core/src/Streamly/Unicode/Stream.hs

```diff
@@ -75,18 +75,25 @@
 --
 module Streamly.Unicode.Stream
     (
+      DecodeState
```

Comment:
Before exposing:

1. Check the naming
2. Check good documentation
3. Check naming consistency
4. Optional: Benchmark and Test